CMakeLists.txt

```diff
@@ -8,6 +8,23 @@ if (${CMAKE_VERSION} VERSION_GREATER 3.1 OR
   cmake_policy(SET CMP0054 NEW)
 endif()
 
+# Set up the project
+project (civetweb)
```

Comment:
project sets all kinds of variables, doesn't it?

Would it be safer to move the CONFIG_CIVETWEB check before 'project'?

Reply:
Well, that would require splitting of platform verification logic into two parts, and effectively duplicating code. `project` is required to have `CMAKE_SYSTEM_NAME`.

By reading the docs and other resources I was led to believe that it shouldn't pollute the rest of the build system. @SebastianBoe Do you know of any super verbose mode in cmake that will help me verify it?

Reply:
You could run cmake with "cmake --trace-expand", and check that it's not doing anything that has unwanted side-effects.

Reply:
You could also consider placing the Zephyr CMake entry-point in another file and then doing

if(CONFIG_CIVETWEB)
include(${CIVETWEB_ROOT}/CMakeLists.txt)
endif()

from there. The entry point is set by the yaml file.

Reply:
The latter option is not something I could do at this moment, but diff on the --trace-expand logs gives me identical outputs. The only thing that differs is:

- temp dir name
- order of files being built (parallel build)
- the actual change in CMakeList I added.

I would consider it side-effect free, do you agree?

Reply:
Yes